fastestimator/pipeline.py

```diff
@@ -163,16 +175,38 @@ def get_modes(self, epoch: Optional[int] = None) -> Set[str]:
             all_modes = set(self.data.keys())
         else:
             all_modes = []
-            for mode, dataset in self.data.items():
+            for mode, datasets in self.data.items():
+                for dataset in datasets.values():
+                    if isinstance(dataset, Scheduler):
+                        dataset = dataset.get_current_value(epoch)
+                    if dataset:
+                        all_modes.append(mode)
+        return to_set(all_modes)
+
+    def get_ds_ids(self, epoch: int, mode: str) -> List[Union[str, None]]:
+        """Get the ds_ids for a given epoch and mode.
+
+        Args:
+            epoch: The current epoch index
+            mode: The current execution mode.
+
+        Returns:
+            The ds_ids of current epoch and mode.
+        """
+        ds_ids = []
+        if mode in self.data:
+            datasets = self.data[mode]
+            for ds_id, dataset in datasets.items():
                 if isinstance(dataset, Scheduler):
```

Comment:
would it make sense to streamline the pipeline member variables by always wrapping inside an epoch scheduler when no scheduler is given?

Reply:
well, I guess not for for now.